packages/backend/src/open_payments/auth/middleware.test.ts

```diff
@@ -266,4 +316,145 @@ describe('Auth Middleware', (): void => {
     expect(ctx.grant).toEqual(grant)
     scope.isDone()
   })
+
+  test('returns 200 with valid http signature', async (): Promise<void> => {
+    Config.skipSignatureVerification = false
+    ctx.request.headers['signature'] = 'sig1=:aGVsbG8=:'
+    ctx.request.headers['signature-input'] =
+      'sig1=("@method" "@target-uri" "content-digest" "content-length" "content-type" "authorization");created=1618884473;keyid="gnap-key"'
+    ctx.request.headers['content-digest'] = 'sha-256=:test-hash:'
+    ctx.request.headers['content-length'] = '1234'
+    const grant = new Grant({
+      active: true,
+      clientId: uuid(),
+      grant: uuid(),
+      access: [
+        {
+          type: AccessType.IncomingPayment,
+          actions: [AccessAction.Read],
+          identifier: ctx.paymentPointer.url
+        }
+      ]
+    })
+    await grantReferenceService.create({
+      id: grant.grant,
+      clientId: grant.clientId
+    })
+    const scope = mockAuthServer(grant.toJSON())
+    await mockIntrospect(grant)
+    await expect(middleware(ctx, next)).resolves.not.toThrow()
+    expect(next).toHaveBeenCalled()
+    scope.isDone()
```

Comment:
```suggestion
    scope.done()
```
- https://github.com/interledger/rafiki/issues/656